client/src/controllers/MessagesController.ts

```diff
@@ -0,0 +1,46 @@
+import { Controller } from '@hotwired/stimulus';
+
+export class MessagesController extends Controller<HTMLFormElement> {
+  static classes = ['newMessage', 'show'];
+  static targets = ['messages', 'templates'];
+
+  declare readonly hasMessagesTarget: boolean;
+  declare readonly hasNewMessageClass: boolean;
+  declare readonly hasShowClass: boolean;
+  declare readonly hasTemplatesTarget: boolean;
+  declare readonly messagesTarget: HTMLElement;
+  declare readonly newMessageClass: string;
+  declare readonly showClass: string;
+  declare readonly templatesTarget: HTMLElement;
+
+  add(event) {
+    const status = event.detail.status;
+    const statusText = event.detail.text;
+    // console.log("connected add.",status,statusText);
+    // console.log(this.templatesTarget);
+    if (this.hasNewMessageClass) {
+      this.element.classList.add(this.newMessageClass);
+    }
+    if (this.hasTemplatesTarget) {
+      const template = this.templatesTarget as HTMLTemplateElement;
+      const cloneTemplate = template.content.cloneNode(true) as HTMLElement;
+      const listElement = cloneTemplate.querySelector(
+        '[data-message-status="' + status + '"]',
+      ) as HTMLElement;
+      if (listElement) {
+        const spanElement = listElement.querySelector('span') as HTMLElement;
+        spanElement.textContent = statusText;
+        if (this.hasMessagesTarget) {
```

Comment:
No need to do this message target check, if the target is not present it will error - that's good in this case. 

Sometimes it's ok for errors to happen if the code is broken